2012-02-18  Aurelien Jarno  <aurelien@aurel32.net>

	* ports/sysdeps/ia64/fpu/e_remainder.S(FREM_X_NAN_INF): Do not return
	an error for remainder(Inf, NaN).
	(FREM_Y_ZERO): Do not return an error for remainder(NaN, 0).
	* ports/sysdeps/ia64/fpu/e_remainderf.S: Likewise.
	* ports/sysdeps/ia64/fpu/e_remainderl.S: Likewise.

---
 ports/sysdeps/ia64/fpu/e_remainder.S  |   14 ++++----------
 ports/sysdeps/ia64/fpu/e_remainderf.S |   14 ++++----------
 ports/sysdeps/ia64/fpu/e_remainderl.S |   14 ++++----------
 3 files changed, 12 insertions(+), 30 deletions(-)

--- a/ports/sysdeps/ia64/fpu/e_remainder.S
+++ b/ports/sysdeps/ia64/fpu/e_remainder.S
@@ -401,12 +401,6 @@
 { .mfi
       nop.m 999
       fclass.m.unc  p8,p0 = f8, 0x23
-      nop.i 999
-}
-// X infinity? Return QNAN indefinite
-{ .mfi
-      nop.m 999
-      fclass.m.unc  p11,p0 = f8, 0x23
       nop.i 999;;
 }
 // Y NaN ?
@@ -429,7 +423,7 @@
 
 {.mfi
       nop.m 999
-(p11) mov f10=f8
+(p8)  mov f10=f8
 	  nop.i 0
 }
 { .mfi
@@ -441,7 +435,7 @@
 { .mfb
       nop.m 999
       frcpa.s0 f8,p7=f8,f9
-	  (p11) br.cond.spnt EXP_ERROR_RETURN;;
+(p8)  br.cond.spnt EXP_ERROR_RETURN;;
 }
 { .mib
 	nop.m 0
@@ -513,10 +507,10 @@
       nop.i 999
 }
 
-{ .mfi
+{ .mfb
       nop.m 999
       fma.d.s0 f8=f11,f1,f0
-      nop.i 999
+(p9)  br.ret.spnt b0 ;;
 }
 
 
--- a/ports/sysdeps/ia64/fpu/e_remainderf.S
+++ b/ports/sysdeps/ia64/fpu/e_remainderf.S
@@ -420,12 +420,6 @@
 { .mfi
       nop.m 999
       fclass.m.unc  p8,p0 = f8, 0x23
-      nop.i 999
-}
-// X infinity? Return QNAN indefinite
-{ .mfi
-      nop.m 999
-      fclass.m.unc  p11,p0 = f8, 0x23
       nop.i 999;;
 }
 // Y NaN ?
@@ -448,7 +442,7 @@
 
 {.mfi
       nop.m 999
-(p11) mov f10=f8
+(p8)  mov f10=f8
 	  nop.i 0
 }
 { .mfi
@@ -460,7 +454,7 @@
 { .mfb
       nop.m 999
       frcpa.s0 f8,p7=f8,f9
-	  (p11) br.cond.spnt EXP_ERROR_RETURN;;
+(p8)  br.cond.spnt EXP_ERROR_RETURN;;
 }
 { .mib
 	nop.m 0
@@ -532,10 +526,10 @@
       nop.i 999
 }
 
-{ .mfi
+{ .mfb
       nop.m 999
       fma.s.s0 f8=f11,f1,f0
-      nop.i 999
+(p9)  br.ret.spnt b0 ;;
 }
 
 
--- a/ports/sysdeps/ia64/fpu/e_remainderl.S
+++ b/ports/sysdeps/ia64/fpu/e_remainderl.S
@@ -427,12 +427,6 @@
 { .mfi
       nop.m 999
       fclass.m.unc  p8,p0 = f8, 0x23
-      nop.i 999
-}
-// X infinity? Return QNAN indefinite
-{ .mfi
-      nop.m 999
-      fclass.m.unc  p11,p0 = f8, 0x23
       nop.i 999;;
 }
 // Y NaN ?
@@ -455,7 +449,7 @@
 
 {.mfi
       nop.m 999
-(p11) mov f10=f8
+(p8)  mov f10=f8
 	  nop.i 0
 }
 { .mfi
@@ -467,7 +461,7 @@
 { .mfb
       nop.m 999
       frcpa.s0 f8,p7=f8,f9
-	  (p11) br.cond.spnt EXP_ERROR_RETURN;;
+(p8)  br.cond.spnt EXP_ERROR_RETURN;;
 }
 { .mib
 	nop.m 0
@@ -542,10 +536,10 @@
       nop.i 999
 }
 
-{ .mfi
+{ .mfb
       nop.m 999
       fma.s0 f8=f11,f1,f0
-      nop.i 999;;
+(p9)  br.ret.spnt b0 ;;
 }
 
 EXP_ERROR_RETURN:
